components/shell/ExternalLabelersContext.tsx

```diff
@@ -61,7 +61,11 @@ export const ExternalLabelersProvider = ({
   // Keep the labelers header up-to-date with the external labelers
   useEffect(() => {
     labelerAgent.configureLabelers([config.did, ...externalLabelers])
-  }, [labelerAgent, config.did, externalLabelers])
+    // If we have an appviewAgent, make sure it respects all configured labelers
+    if (appviewAgent) {
+      appviewAgent.configureLabelers([config.did, ...externalLabelers])
+    }
+  }, [labelerAgent, appviewAgent, config.did, externalLabelers])
```

Comment:
I would split this into two distinct effects. Indeed, with the current implementation, if `appviewAgent` is updated, but `labelerAgent` isn't, `labelerAgent` will be re configured anyways. This is very much a nit in this particular case, but a good practice in general (avoiding un-necessary side effects).